jax/_src/config.py

```diff
@@ -1555,13 +1555,11 @@ def _update_x64_thread_local(val):
 def _update_default_device_global(val):
   lib.jax_jit.global_state().default_device = val
 
-
 def _update_default_device_thread_local(val):
   lib.jax_jit.thread_local_state().default_device = val
 
-
 def _validate_default_device(val):
-  if val is not None and not isinstance(val, xla_client.Device):
+  if val is not None and not isinstance(val, xla_client.Device) and val not in ['cpu', 'gpu', 'tpu']:
```

Comment:
I am not sure if this is a good idea. Parameterizing default device on platform name doesn't sound like the right API.

What problem is this solving?

Reply:
This change was my idea. Stella is onboarding on my new team and I thought this would be a good first PR haha.

The problem this is solving is being able to quickly switch to host computation. I think this was the original point of the default_device API (although it can be used to switch between accelerators too). In addition, this lets you set the default device to host on the command line (JAX_DEFAULT_DEVICE=cpu). I meant to expand it to take platform strings after landing the initial change (I even left a TODO), but that never happened...

Do you have concerns with this change? It's really just making the API more ergonomic (`"cpu"` instead of `jax.devices("cpu")[0]`), not fundamentally changing what it can do.

Reply:
`JAX_PLATFORMS=cpu` does something slightly different: it makes it so the default TPU backend isn't even available. JAX_DEFAULT_DEVICE still initializes all the backends, it just changes the device that single-device operations run on if no device is otherwise specified. I'm not sure how many applications will actually want to globally set the default device to CPU (that's why I think the main benefit is better ergonomics for the context manager), but it still makes it possible in case anyone does.

I don't agree that users should always need to type in the long `jax.devices(platform)[device_no]` for consistency even if they don't actually care about which device is it (or in the common case of CPU, there's only one device usually). We should provide ergonomic APIs for common patterns.

Reply:
The pattern you are providing is inconsistent though. I think we should spend some time thinking about this because choosing a default device based on platform name does not seem like the obvious choice here.